spec/notifications/user_update_notification_spec.rb

```diff
@@ -0,0 +1,21 @@
+require "rails_helper"
+
+RSpec.describe(UserUpdateNotification, type: :model) do
+  let!(:user) { create(:user) }
+
+  context "when a user is updated" do
+    it "creates database notification" do
+      expect { user.update(email: "new@email.com") }.to(change(Notification, :count).by(1))
```

Comment:
Interesting, we've never used this style of implementation before! `change(Notification, :count)`. My only issue is that the mailer check is different than this, with `UserMailer.deliveries.count` being the implementation. Personally I prefer the latter, but its up to you!

Reply:
My intent was to check _both_ `database` style notifications, and `mailer` style notifications, hence the different approaches to 'counting' them. But maybe it's not necessary to test for both?

Reply:
Checking both seems reasonable, and I get the two strategies. I think we're good to proceed here!